Animations/vendor/CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+# Vendored libraries
+
+# This file acts as a proxy between all the vendored libraries
+# and the CMake build system.
+
+###############
+# CMake setup #
+###############
+include(ExternalProject)
+
+############
+# cppUtils #
+############
+add_library(CppUtils INTERFACE)
+target_include_directories(CppUtils INTERFACE "${CMAKE_CURRENT_LIST_DIR}/cppUtils/single_include")
+
+########
+# GLFW #
+########
+set(GLFW_BUILD_DOCS OFF CACHE BOOL "" FORCE)
+set(GLFW_BUILD_TESTS OFF CACHE BOOL "" FORCE)
+set(GLFW_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
+add_subdirectory(GLFW)
+
+############
+# Freetype #
+############
+add_subdirectory(freetype)
+
+##########
+# OpenAL #
+##########
+set(ALSOFT_UTILS OFF)
+set(ALSOFT_NO_CONFIG_UTIL ON)
+set(ALSOFT_EXAMPLES OFF)
+set(ALSOFT_INSTALL OFF)
+set(ALSOFT_INSTALL_CONFIG OFF)
+set(ALSOFT_INSTALL_HRTF_DATA OFF)
+set(ALSOFT_INSTALL_AMBDEC_PRESETS OFF)
+set(ALSOFT_INSTALL_EXAMPLES OFF)
+set(ALSOFT_INSTALL_UTILS OFF)
+add_subdirectory(openal)
+
+########
+# Luau #
+########
+set(CMAKE_CXX_STANDARD 17) # Luau doesn't build with C++20
+
+set(LUAU_BUILD_CLI OFF)
+set(LUAU_BUILD_TESTS OFF)
+add_subdirectory(luau)
+
+set(CMAKE_CXX_STANDARD 20) # See above
+
+#############
+# Onigurama #
+#############
+add_subdirectory(onigurama)
+
+############
+# TinyXML2 #
+############
+add_subdirectory(tinyxml2)
+
+###########
+# PlutoVG #
+###########
+add_subdirectory(plutovg)
```

Comment:
Adding the entire plutovg subdirectory adds in unnecessary test projects into the workspace. I did this to only add the source code and was wondering if we can do something similar?

```cmake
add_library(plutovg STATIC)

add_subdirectory(Animations/vendor/plutovg/include)
add_subdirectory(Animations/vendor/plutovg/source)
```

Reply:
I think this would work in theory, but I can only see one extra target being added which sadly can't be disabled. It might be worth considering PR'ing the wanted changes upstream - this avoids future breakage and also benefits other people using the library.

Reply:
I don't have the time or patience to submit a PR to the original repository at this moment, I'm fine with adding the subdirectories manually like this for now. If, and when, I have to upgrade any dependencies, there will likely be breaking changes anyways that I'll have to thoroughly test, so having the build potentially break after an upgrade doesn't bother me anyways. I also don't take upgrades lightly because I've been burned quite a few times in the past, so whenever I do upgrade a dependency I usually have to spend thorough time reviewing the changes and making sure the upgrade is safe.